Unfriend Variable factory functions.

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #28620 Merge Tensor and Variable.
* #28610 Null AutogradMeta optimization
* #28609 Delete dead TensorImpl::detach_autograd_meta
* #28602 Remove unused gradient_edge argument from make_variable_view
* #28593 Add an AutogradMeta factory.
* **#28601 Unfriend Variable factory functions.**
* #28592 Move all autograd_meta_ manipulating operations out-of-line.
* #28543 Test TensorTypeSet instead of autograd_meta_ for variable-ness.

In the process, I moved AutogradMeta out of the Variable class. The
intent here is that I'm going to delete Variable class entirely,
so I had better not be putting stuff in it!

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D18171160](https://our.internmc.facebook.com/intern/diff/D18171160)

ezyang committed:
Unfriend Variable factory functions.

In the process, I moved AutogradMeta out of the Variable class. The
intent here is that I'm going to delete Variable class entirely,
so I had better not be putting stuff in it!

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang commented:
Sorry, the old PR at #28282 got accidentally closed by ghstack (I need to fix this lol)

ezyang committed:
Update on "Unfriend Variable factory functions."


In the process, I moved AutogradMeta out of the Variable class. The
intent here is that I'm going to delete Variable class entirely,
so I had better not be putting stuff in it!

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Unfriend Variable factory functions."


In the process, I moved AutogradMeta out of the Variable class. The
intent here is that I'm going to delete Variable class entirely,
so I had better not be putting stuff in it!

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Unfriend Variable factory functions."


In the process, I moved AutogradMeta out of the Variable class. The
intent here is that I'm going to delete Variable class entirely,
so I had better not be putting stuff in it!

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D18171160](https://our.internmc.facebook.com/intern/diff/D18171160)

[ghstack-poisoned]

ezyang committed:
Update on "Unfriend Variable factory functions."


In the process, I moved AutogradMeta out of the Variable class. The
intent here is that I'm going to delete Variable class entirely,
so I had better not be putting stuff in it!

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D18171160](https://our.internmc.facebook.com/intern/diff/D18171160)

[ghstack-poisoned]

ezyang committed:
Update on "Unfriend Variable factory functions."


In the process, I moved AutogradMeta out of the Variable class. The
intent here is that I'm going to delete Variable class entirely,
so I had better not be putting stuff in it!

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D18171160](https://our.internmc.facebook.com/intern/diff/D18171160)

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit d2e44e0:
* 0/1 flaky



Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=d2e44e077de2eaf07657cb73778bdf008b552230).

---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

ezyang committed:
Update on "Unfriend Variable factory functions."


In the process, I moved AutogradMeta out of the Variable class. The
intent here is that I'm going to delete Variable class entirely,
so I had better not be putting stuff in it!

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D18171160](https://our.internmc.facebook.com/intern/diff/D18171160)

[ghstack-poisoned]
